docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,275 @@
+
+Substreams modules are created and prepared for the sink tool. After the sink tool begins running, automated ingestion tasks can be setup to have BigQuery, or other files-based storage solution, ingest the data.
+
+As an example, you could use `substreams-sink-files` to sink data in `JSONL` format to a [Google Cloud Storage (GCS)](https://cloud.google.com/storage) bucket and configure a BigQuery Transfer job to run every 15 minutes. The scheduled job ingests the new files found in the GCS bucket where the data, extracted by the Substreams, was written.
+
+### Accompanying code example
+
+[The accompanying Substreams module](#) associated with this documentation is responsible for extracting a handful of data fields from the Block object injected into the Rust-based map module. The sink tool processes the extracted blockchain data line-by-line and outputs the data to the files-based persistence mechanism you've chosen.
+
+The accompanying code example extracts four data points from the Block object and packs them into the `substreams.sink.files.v1` protobuf's data model. The data is passed to the protobuf as a single line of plain text, a `CSV` entry, or a `JSONL` element. StreamingFast is working on support for binary formats such as [Avro](https://avro.apache.org/), [Parquet](https://parquet.apache.org/), and others.
+
+## Outline
+
+1. Download code and tools
+2. Run code and verify output
+3. Code walkthrough
+4. Understanding the sink tool
+5. Run tool and verify output
+6. Conclusion and review
+7. Getting help
+
+## Download code and tools
+
+### Install `substreams-sink-files`
+
+**TODO:** <i>Where is the actual binary file? What is the file type/format/extension (if any)?</i>
+https://github.com/streamingfast/substreams-sink-files/releases
+
+### Verify installation
+
+Checking the version of `substreams-sink-files` will produce a message similar to:
+
+```bash
+substreams-sink-files version dev
+```
+
+### Update system `PATH`
+
+If you want to use the `substreams-sink-tool` from anywhere on your system make sure the computer's `PATH` environment variable has been set correctly.
+
+Add the following lines to the computer's `~/.bashrc` configuration file and then restart the shell session.
+
+```bash
+export GOPATH=$HOME/go
+export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+```
+
+### Download accompanying code example
+
+The accompanying code example is available in its [official GitHub repository](#). You'll need to download or clone the repository to your computer to complete this tutorial.
+
+You can use the `git clone` command to acquire the example code.
+
+```bash
+git clone TODO_INSERT_REPO_LINK_WHEN_AVAILABLE
+```
+
```

Comment:
Explanation of what is required from Substreams module perspective. We need to talk here about the Protobuf gen required to pull `sf.substreams.sink.files.v1` model and modifications required, e.g. outputting a JSON model for each entity, one entity per line.

Reply:
Noted in file. Need additional input.

Reply:
It mainly a description of the code at https://github.com/streamingfast/substreams-sink-files/blob/master/docs/tutorial/src/lib.rs#LL8-L22C2

- You need to import sink files `.spkg` and re-generate proto (with a note that people don't forget to add it to their `mod.rs`
- You then need to create a `map` module that will output `sf.substreams.sink.files.v1` format.
- This module extract from the block (or from other module's dependencies) the entity to be written, one per block.
- Each line will be in JSON format in our example, you can use `serde_json` `json!` macro to quickly create your structure, one per line.
- Add to you `substreams.yaml` the correct module definition, here it is (show sample part of the module).

Mostly something like that, essential describing what a user needs to do in their Substreams to output for sink files.

Reply:
This is addressed.